instagram/src/components/Authentication/Authenticate.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import Login from '../Login/Login';
+
+const Authenticate = App =>
+  class extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        isLoggedIn: false
+      };
+    }
+
+    componentDidMount() {
+      if (localStorage.getItem('user')) {
+        this.setState({ isLoggedIn: false });
+      } else {
+        this.setState({ isLoggedIn: true });
+      }
+    }
+
+    render() {
+      if (this.state.isLoggedIn) return <Login />;
```

Comment:
the logic seems a little reversed ... it looks like if `isLoggedIn is true ... you show the <Login/> component but logically you would want <Login/> to be returned if isLoggedIn is false. It works ... just reads funny.